From de819ca8136f2a5806c7816ee1c7b82f73643f3f Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@notvalid.com>
Date: Wed, 14 Oct 2026 16:38:31 -0700
Subject: [PATCH] Keep historical Scrypt header vectors with current stream API

Replace the removed CDataStream test constructor without changing genesis, Digishield or merge-mining boundary serialization and PoW vectors.

diff --git a/src/test/baseheader_tests.cpp b/src/test/baseheader_tests.cpp
index 597f792b3a81ced4fef1a05f07db878df25295c2..b49f212b3cc09b01ca382252d6ad20dff75378ab 100644
--- a/src/test/baseheader_tests.cpp
+++ b/src/test/baseheader_tests.cpp
@@ -27,7 +27,7 @@ BOOST_AUTO_TEST_CASE(genesis) {
         "0000026f3f7874ca0c251314eaed2d2fcf83d7da3acfaacf59417d485310b448");
 
     // Check serialization
-    CDataStream ss(0, 0);
+    DataStream ss{};
     ss << genesis;
     BOOST_CHECK_EQUAL(HexStr(ss),
                       "01000000000000000000000000000000000000000000000000000000"
@@ -53,7 +53,7 @@ BOOST_AUTO_TEST_CASE(block_145000) {
         "00000000002dfb87dd0d1b359eac948f33e91f87d586d36c497df2b08db7eb8a");
 
     // Check serialization
-    CDataStream ss(0, 0);
+    DataStream ss{};
     ss << block;
     BOOST_CHECK_EQUAL(HexStr(ss),
                       "0200000058054081d6f4a30d6976ac03be7e3890f67fd8331613bb7a"
